src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7530,37 +7515,54 @@ private BoundExpression MakeMemberAccessValue(BoundExpression expr, BindingDiagn
+        //
+        // - if the extension member lookup finds nothing, then we don't need to touch the result for the member access.
+        internal BoundExpression? ResolveExtensionMemberAccessIfResultIsNonMethod(SyntaxNode syntax, BoundExpression receiver, string name,
+          SeparatedSyntaxList<TypeSyntax> typeArgumentsSyntax, ImmutableArray<TypeWithAnnotations> typeArgumentsOpt,
+            BindingDiagnosticBag diagnostics)
         {
-            switch (expr.Kind)
-            {
-                case BoundKind.MethodGroup:
-                    if (expr is BoundMethodGroup { Methods: [], SearchExtensionMethods: true } methodGroup)
-                    {
-                        CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
-                        var resolution = this.ResolveMethodGroup(methodGroup, analyzedArguments: null, ref useSiteInfo, options: OverloadResolution.Options.None);
-                        diagnostics.Add(expr.Syntax, useSiteInfo);
+            CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+            // Note: we're resolving without arguments, which means we're not treating the member access as invoked
+            var resolution = this.ResolveExtension(
+                syntax, name, analyzedArguments: null, receiver, typeArgumentsOpt, options: OverloadResolution.Options.None,
+                returnRefKind: default, returnType: null, withDependencies: useSiteInfo.AccumulatesDependencies);
 
-                        if (resolution.IsExtensionMember(out Symbol extensionMember))
-                        {
-                            diagnostics.AddRange(resolution.Diagnostics);
-                            resolution.Free();
-                            return GetExtensionMemberAccess(methodGroup.Syntax, methodGroup.ReceiverOpt, extensionMember,
-                                methodGroup.TypeArgumentsSyntax, methodGroup.TypeArgumentsOpt, diagnostics);
-                        }
+            diagnostics.Add(syntax, useSiteInfo);
 
-                        resolution.Free();
-                    }
+            if (resolution.IsNonMethodExtensionMember(out Symbol? extensionMember))
+            {
+                diagnostics.AddRange(resolution.Diagnostics);
+                resolution.Free();
 
-                    return expr;
-                // PROTOTYPE need to handle PropertyGroup as well
+                var memberAccess = GetExtensionMemberAccess(syntax, receiver, extensionMember,
+                    typeArgumentsSyntax, typeArgumentsOpt, diagnostics);
 
-                default:
-                    return expr;
+                if (!memberAccess.HasErrors && typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument))
+                {
+                    Error(diagnostics, ErrorCode.ERR_OmittedTypeArgument, syntax);
```

Comment:
>Error(diagnostics, ErrorCode.ERR_OmittedTypeArgument, syntax); [](http://example.com/codeflow?start=20&length=62)

Do we have a test covering this code path? Including scenario when the member is a type

Reply:
Yes, `ExtensionMemberLookup_MatchingExtendedType_GenericType_GenericMember_OmittedTypeArgument` (a generic type scenario) and `ExtensionInvocation_TypeReceiver_TypeArguments_Omitted` and `ExtensionInvocation_InstanceReceiver_TypeArguments_Omitted` (a generic method, with either a type or an instance receiver).